airflow/utils/helpers.py

```diff
@@ -202,3 +204,12 @@ def cross_downstream(*args, **kwargs):
         stacklevel=2,
     )
     return import_string('airflow.models.baseoperator.cross_downstream')(*args, **kwargs)
+
+
+def is_k8s_or_k8scelery_executor() -> bool:
+    """Determines if the executor utilizes k8s."""
+    conf_executor = conf.get('core', 'EXECUTOR')
+    return conf_executor in {
+        executor_constants.KUBERNETES_EXECUTOR,
+        executor_constants.CELERY_KUBERNETES_EXECUTOR,
+    }
```

Comment:
Wouldn't it be better to move it to `executor_constants` as `IS_K8S_OR_CELERY = conf.get('core', 'EXECUTOR') in CELERY_OR_K8S_EXECUTORS`?

Reply:
an `Is_K8S_OR_CELERY` feels like it would fit well in `airflow.settings` -- but doing that without an import cycle might be tricky.

Reply:
Updated in https://github.com/apache/airflow/pull/11815/commits/aaa04916eef7437dd2ebde339c9d7efacef72dbc